common/utils_unittest.cpp

```diff
@@ -126,9 +126,9 @@ UTILS_TEST(guessResolutionOverflow) {
             << "x"
             << std::numeric_limits<long>::max();
 
-    EXPECT_FALSE(guessResolution(sstream.str().c_str(), w, h));
-    EXPECT_EQ(w, 0);
-    EXPECT_EQ(h, 0);
+    EXPECT_TRUE(guessResolution(sstream.str().c_str(), w, h));
```

Comment:
This test case should not be changed.  It exposes a bug in `guessResolution`.  Please fix `guessResolution` instead.  That is, `guessResolution` overflows the **int** output parameters (w,h) if it parses a width or height from `filename` param that is larger than what can fit in an `int` value.  This is the bug.

As @xuguangxin pointed out, `guessResolution` should avoid overflow and just make w,h = 0 instead.